analyzers/src/SonarAnalyzer.Common/Json/Parsing/LexicalAnalyzer.cs

```diff
@@ -264,8 +264,8 @@ private object ReadNumberValue()
         {
             StringBuilder @decimal = null;
             StringBuilder exponent = null;
-            var integer = new StringBuilder();
-            var current = integer;
+            var number = new StringBuilder();
```

Comment:
The parser should follow the json spec for numbers https://www.json.org/json-en.html
```
number
  integer fraction exponent

integer
  digit
  onenine digits
  '-' digit
  '-' onenine digits

digits
  digit
  digit digits

digit
  '0'
  onenine

onenine
  '1' . '9'

fraction
  ""
  '.' digits

exponent
  ""
  'E' sign digits
  'e' sign digits

sign
  ""
  '+'
  '-'
```

JSON numbers are unlimited in their size. It is a good compromise to use `double` for JSON numbers.
You can find more info here https://stackoverflow.com/a/13502497

Reply:
SGTM - and makes things simpler! thanks!